.circleci/config.yml

```diff
@@ -59,21 +65,144 @@ commands:
           command: circleci tests glob "**/test*/**/*.kt" | circleci tests split | xargs -n 1 echo
       - run:
           name: << parameters.desc >>
-          command: ./gradlew << parameters.cmd >> -i -PtestFilter="`circleci tests glob "**/test*/**/*.kt" | circleci tests split`"
+          command: >
+            ./gradlew -PdisablePreDex
+            << parameters.cmd >>
+            -i -PtestFilter="`circleci tests glob "**/test*/**/*.kt" | circleci tests split`"
           environment:
-            JVM_OPTS: -Xmx2048m
-            GRADLE_OPTS: -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError -Dorg.gradle.caching=true -Dorg.gradle.configureondemand=true -Dkotlin.compiler.execution.strategy=in-process -Dkotlin.incremental=false
+            JVM_OPTS: -Xmx4096m
+            GRADLE_OPTS: >
+              -Xmx1536m -XX:+HeapDumpOnOutOfMemoryError
+              -Dorg.gradle.caching=true
+              -Dorg.gradle.configureondemand=true
+              -Dkotlin.compiler.execution.strategy=in-process
+              -Dkotlin.incremental=false
+
+  skip-for-external-pull-requests:
```

Comment:
I think this also is not needed, if we run it only for release build, right?

Reply:
We need this to skip sonar for external pull requests. Otherwise the sonar check will fail and external PRs can only be merged with admin permission. By skipping the check and ending it with a positive result, others can merge external PRs too. Not ideal that sonarr is skipped but also not ideal that only admins can merge external PRs.